tests/tx/scripts/test_sighash.py

```diff
@@ -0,0 +1,321 @@
+        self.manager2: HathorManager = self.create_peer('testnet', unlock_wallet=True, wallet_index=True)
+
+        # 1 is Alice
+        assert self.manager1.wallet
+        self.address1_b58 = self.manager1.wallet.get_unused_address()
+        self.private_key1 = self.manager1.wallet.get_private_key(self.address1_b58)
+        self.address1 = decode_address(self.address1_b58)
+
+        # 2 is Bob
+        assert self.manager2.wallet
+        self.address2_b58 = self.manager2.wallet.get_unused_address()
+        self.address2 = decode_address(self.address2_b58)
+
+        self.genesis_private_key = get_genesis_key()
+        self.genesis_block = self.manager1.tx_storage.get_transaction(self._settings.GENESIS_BLOCK_HASH)
+
+        # Add some blocks so we can spend the genesis outputs
+        add_blocks_unlock_reward(self.manager1)
+
+    @patch('hathor.transaction.scripts.opcode.is_opcode_valid', lambda _: True)
+    def test_sighash_bitmask(self) -> None:
+        # Create a new test token
+        token_creation_tx = create_tokens(self.manager1, self.address1_b58)
+        token_uid = token_creation_tx.tokens[0]
+        token_creation_utxo = token_creation_tx.outputs[0]
+        genesis_utxo = self.genesis_block.outputs[0]
+        parents = self.manager1.get_new_tx_parents()
+
+        # Alice creates an input spending all created test tokens
+        tokens_input = TxInput(not_none(token_creation_tx.hash), 0, b'')
+
+        # Alice creates an output sending half genesis HTR to herself
+        alice_output_script = P2PKH.create_output_script(self.address1)
+        htr_output = TxOutput(int(genesis_utxo.value / 2), alice_output_script)
+
+        # Alice creates an atomic swap tx that's missing only Bob's input, with half genesis HTR
+        atomic_swap_tx = Transaction(
+            weight=1,
+            inputs=[tokens_input],
+            outputs=[htr_output],
+            parents=parents,
+            tokens=[token_uid],
+            storage=self.manager1.tx_storage,
+            timestamp=token_creation_tx.timestamp + 1
+        )
+        self.manager1.cpu_mining_service.resolve(atomic_swap_tx)
+
+        # Alice signs her input using sighash bitmasks, instead of sighash_all.
+        sighash_bitmask = SighashBitmask(inputs=0b1, outputs=0b1)
+        data_to_sign1 = atomic_swap_tx.get_custom_sighash_data(sighash_bitmask)
+        assert self.manager1.wallet
+        public_bytes1, signature1 = self.manager1.wallet.get_input_aux_data(data_to_sign1, self.private_key1)
+        tokens_input.data = P2PKH.create_input_data(
+            public_key_bytes=public_bytes1,
+            signature=signature1,
+            sighash=sighash_bitmask,
+        )
+
+        # At this point, the tx is partial. The inputs are valid, but they're mismatched with outputs
+        self.manager1.verification_service.verifiers.tx.verify_inputs(atomic_swap_tx)
```

Comment:
Shouldn't it fail since there's no sighash all?

Reply:
This requirement is only implemented in Part 7: https://github.com/HathorNetwork/hathor-core/pull/917